src/instana/instrumentation/sanic_inst.py

```diff
@@ -1,144 +1,125 @@
+                span.set_attribute("http.path", request.path)
+                span.set_attribute(SpanAttributes.HTTP_METHOD, request.method)
+                span.set_attribute(SpanAttributes.HTTP_HOST, request.host)
                 if hasattr(request, "url"):
-                    scope.span.set_tag("http.url", request.url)
+                    span.set_attribute(SpanAttributes.HTTP_URL, request.url)
 
                 query = request.query_string
 
                 if isinstance(query, (str, bytes)) and len(query):
                     if isinstance(query, bytes):
-                        query = query.decode('utf-8')
-                    scrubbed_params = strip_secrets_from_query(query, agent.options.secrets_matcher,
-                                                               agent.options.secrets_list)
-                    scope.span.set_tag("http.params", scrubbed_params)
-
-                if agent.options.extra_http_headers is not None:
-                    extract_custom_headers(scope.span, headers)
-                await wrapped(*args, **kwargs)
+                        query = query.decode("utf-8")
+                    scrubbed_params = strip_secrets_from_query(
+                        query, agent.options.secrets_matcher, agent.options.secrets_list
+                    )
+                    span.set_attribute("http.params", scrubbed_params)
+
+                if agent.options.extra_http_headers:
+                    extract_custom_headers(span, headers)
                 if hasattr(request, "uri_template") and request.uri_template:
-                    scope.span.set_tag("http.path_tpl", request.uri_template)
-                if hasattr(request, "ctx"):  # ctx attribute added in the latest v19 versions
-                    request.ctx.iscope = scope
-        except Exception as e:
-            logger.debug("Sanic framework @ handle_request", exc_info=True)
-            return await wrapped(*args, **kwargs)
-
-
-    logger.debug("Instrumenting Sanic")
+                    span.set_attribute("http.path_tpl", request.uri_template)
+            except Exception:
+                logger.debug("request_with_instana: ", exc_info=True)
+
+        @app.exception(Exception)
+        def exception_with_instana(request: Request, exception: Exception) -> None:
+            try:
+                if not hasattr(request.ctx, "span"): # pragma: no cover
+                    return
+                span = request.ctx.span
+
+                if isinstance(exception, SanicException):
+                    # Handle Sanic-specific exceptions
+                    status_code = exception.status_code
+                    message = str(exception)
+
+                    if all([span, status_code, message]) and 500 <= status_code:
+                        span.set_attribute("http.error", message)
+            except Exception:
+                logger.debug("exception_with_instana: ", exc_info=True)
```

Comment:
Shouldn't we record exception to the span here?

```suggestion
        except Exception as exc:
                span.record_exception(exc)
                logger.debug("exception_with_instana: ", exc_info=True)
```

Reply:
https://github.com/instana/python-sensor/pull/613#discussion_r1756901926